routers/web/repo/editor.go

```diff
@@ -80,6 +82,21 @@ func redirectForCommitChoice(ctx *context.Context, commitChoice, newBranchName,
 		}
 	}
 
+	prIndex := ctx.FormInt64("back_to_pr")
+	if prIndex > 0 {
+		// Redirect to the PR's files tab
+		link := ctx.Repo.RepoLink + fmt.Sprintf("/pulls/%d/files?refresh=%d", prIndex, time.Now().Unix())
+		diffHash := ctx.FormString("back_to_diff_hash")
```

Comment:
`back_to_pr` + `back_to_diff_hash` is quite fragile. I think it should use a prepared `return_uri`

Reply:
I would like to control the return_uri's scope. It should not be redirected to other places currently.

Reply:
Why "other places" would happen? There is a function `RedirectToFirst` to handle "redirect_uri".

Reply:
06fb88b8c95be465b3bf7fec11f8a0c56b4447a3